### ISSUE TITLE
Question about Max, Min Depth Value

### ISSUE THREAD

**Issue by ssh10032** [shariqfarooq123/AdaBins#68]
Hi, I'm going to train Adabins to my custom dataset. 
At first, there is a part where the maximum and minimum values are specified in the part where the model is loaded, does this have an effect on training? 
`model = UnetAdaptiveBins.build(n_bins=256, min_val=0, max_val=2.5, norm='linear').to(device)`
And is it a problem if I use these values differently when I load the weights later?

Thank you for providing your work. 
It is very helpful for studying.

**Comment by shariqfarooq123** [comment]
Hi, 

Thanks for your interest in our work. 
I hypothesize that changing `min_val` and `max_val` on pretrained model would just change units the output depth map uses. By default, the units are meters but if you change the `max_val` to `new_max_val`, then output units would be scaled by ` (new_max_val - new_min_val) / (max_val - min_val)`.

I'm curious if you experimented with this and your concluding insights about this situation.

**Comment by Tram-Tran** [comment]
Hi,

I downloaded and read the predicted depth maps on the KITTI test set that you provide in the README. I see that the max and min values are about 20000 and < 1000. Are they in meters or millimeters?

Thank you for providing your work.

**Comment by LeTrongDuc2510** [comment]
> Hi,
> 
> I downloaded and read the predicted depth maps on the KITTI test set that you provide in the README. I see that the max and min values are about 20000 and < 1000. Are they in meters or millimeters?
> 
> Thank you for providing your work.


In dataloader.py, for NYU, the pixel values are divided by 1000, meaning they are in millimeters. For KITTI, the pixel values are divided by 256, which is not in meters or millimeters. This is just a convenient way to store depth values since all depth maps are in uint16 format, where the maximum value is 65535. However, KITTI requires storing depths up to 80m (80000mm), so the choice of depth units depends on the image format and the depth range of each dataset. Hope this helps!
